orbbec_camera/src/ros_service.cpp

```diff
@@ -1069,6 +1072,63 @@ void OBCameraNode::resetCaptureServiceVariables() {
   number_of_depth_frames_captured_ = 0;
 }
 
+void OBCameraNode::handleChangeStateRequest(
+    const std::shared_ptr<lifecycle_msgs::srv::ChangeState::Request> request,
+        std::shared_ptr<lifecycle_msgs::srv::ChangeState::Response> response) {
+    
+  RCLCPP_INFO(logger_, "Received request to change state '%d' with label '%s'",
+                     request->transition.id, request->transition.label.c_str());
+  
+  response->success = true;
+  if(request->transition.id == lifecycle_msgs::msg::Transition::TRANSITION_ACTIVATE) {
+    RCLCPP_INFO_STREAM(logger_, "Recieved request to turn ON camera streams");
+    if (pipeline_started_){
+      RCLCPP_WARN_STREAM(logger_, "Camera streams already ON");
+      response->success = false;
+      return;
+    }
+    try {
+      setupProfiles();
+      startStreams();
+      RCLCPP_INFO_STREAM(logger_, "Camera streams are now ON");
+    } catch (const ob::Error& e) {
+      response->success = false;
+      RCLCPP_ERROR_STREAM(logger_, "Failed to start camera streams: " << e.getMessage());
+    } catch (const std::exception& e) {
+      response->success = false;
+      RCLCPP_ERROR_STREAM(logger_, "Failed to start camera streams: " << e.what());
+    } catch (...) {
+      response->success = false;
+      RCLCPP_ERROR_STREAM(logger_, "Failed to start camera streams: Unknown Error");
+    }
+  }
+  else if(request->transition.id == lifecycle_msgs::msg::Transition::TRANSITION_DEACTIVATE) {
+    RCLCPP_INFO_STREAM(logger_, "Recieved request to turn OFF camera streams");
+    if (!pipeline_ || !pipeline_started_){
+      RCLCPP_WARN_STREAM(logger_, "Camera streams already OFF");
+      response->success = false;
```

Comment:
```suggestion
      response->success = true;
```
this'd make it a successful but no-op transition, I think it works better for the clients of this service